### ISSUE TITLE
Which optimizer we use in yolov3

### ISSUE THREAD

**Issue by YCAyca** [AlexeyAB/darknet#6380]
-------------
learning_rate=0.001
burn_in=1000
max_batches = 10000
policy=steps
steps=8000,9000
scales=.1,.1
-----------------------

Hi, does anyone know which optimizer is used in Yolov3? Because I understand what the burn_in, initial learning rate and the policy on the .cfg file and how to update the learning rate but dont understand the l.update(l, update_batch,rate,net.decay) function called from update_network(...) function.

![image](https://user-images.githubusercontent.com/20625822/88930204-d12a9900-d283-11ea-83c9-7179f400696f.png)

That's why I don't understand what the decay=0.0005 parameter in .cfg file means. I assume that decay parameter is used to reupdate the learning rate coming from get_current_rate() function and then we use this learning rate in l.update() function to update the weights using an optimizer. So the question is if this assumption is true? If true, which optimization do we use and how can I see the definition of l.update(....) line? Because when I right click to go to its definition, I can't because it's not a function what VS says.

![image](https://user-images.githubusercontent.com/20625822/88930565-5150fe80-d284-11ea-944e-313f9d6cfc0d.png)

**Comment by WongKinYiu** [comment]
optimizer: SGD + momentum + weight decay
```
momentum=0.9
decay=0.0005
```

learning rate scheduler: burn in + multi-steps decay
```
learning_rate=0.001
burn_in=1000
max_batches = 500200
policy=steps
steps=400000,450000
scales=.1,.1
```

so the decay in cfg means weight decay, not learning rate decay.
